packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -1365,6 +1387,62 @@ describe('TokensController', () => {
 
       generateRandomIdStub.mockRestore();
     });
+
+    it('stores multiple tokens from a batched watchAsset confirmation screen correctly when user confirms', async function () {
+      const generateRandomIdStub = jest
+        .spyOn(tokensController, '_generateRandomId')
+        .mockImplementationOnce(() => requestId)
+        .mockImplementationOnce(() => '67890');
+
+      const acceptedRequest = new Promise<void>((resolve) => {
+        tokensController.subscribe((state) => {
+          if (
+            state.allTokens?.[ChainId.mainnet]?.[interactingAddress].length ===
+            2
+          ) {
+            resolve();
+          }
+        });
+      });
+
+      const anotherAsset = {
+        address: '0x000000000000000000000000000000000000ABcD',
+        decimals: 18,
+        symbol: 'TEST',
+        image: 'image2',
+        name: undefined,
+      };
+
+      tokensController.watchAsset({ asset, type, interactingAddress });
+      tokensController.watchAsset({
+        asset: anotherAsset,
+        type,
+        interactingAddress,
+      });
+
+      await approvalController.accept(requestId);
+      await approvalController.accept('67890');
+      await acceptedRequest;
+
+      expect(
+        tokensController.state.allTokens[ChainId.mainnet][interactingAddress],
+      ).toHaveLength(2);
+      expect(
+        tokensController.state.allTokens[ChainId.mainnet][interactingAddress],
+      ).toStrictEqual([
+        {
+          isERC721: false,
+          aggregators: [],
+          ...asset,
+        },
+        {
+          isERC721: false,
+          aggregators: [],
+          ...anotherAsset,
+        },
+      ]);
+      generateRandomIdStub.mockRestore();
```

Comment:
`jest.resetAllMocks()` in a `beforeEach()`?

EDIT: ohh... I see. This pattern is used 4 other times in this file already. Up to you if you want to clean it up or not